### ISSUE TITLE
Traceback on refreshing data

### ISSUE THREAD

**Issue by damazter** [toolsforexperiments/plottr#75]
When refreshing the data, sometimes plottr gives a traceback similar to the following form

```python
Traceback (most recent call last):
  File "plottr\plottr\apps\inspectr.py", line 425, in plotRun
    fc, win = autoplotQcodesDataset(pathAndId=(self.filepath, runId))
  File "plottr\plottr\apps\autoplot.py", line 238, in autoplotQcodesDataset
    win = QCAutoPlotMainWindow(fc, pathAndId=pathAndId)
  File "plottr\plottr\apps\autoplot.py", line 213, in __init__
    self.setDefaults()
  File "plottr\plottr\apps\autoplot.py", line 182, in setDefaults
    self.fc.nodes()['Dimension assignment'].dimensionRoles = drs
  File "plottr\plottr\node\node.py", line 40, in wrap
    self.ui.optSetters[optName](val)
  File "plottr\plottr\node\dim_reducer.py", line 627, in setRoles
    self.widget.setRole(dimName, role)
  File "plottr\plottr\node\dim_reducer.py", line 336, in setRole
    super().setRole(dim, role, **kw)
  File "plottr\plottr\node\dim_reducer.py", line 276, in setRole
    super().setRole(dim, role)
  File "plottr\plottr\node\dim_reducer.py", line 202, in setRole
    self.choices[dim]['roleSelectionWidget'].setCurrentText(role)
KeyError: 'NW_bias'
```

Where `NW_bias` is one of the parameters that i am measuring

**Comment by mgunyho** [comment]
Is `NW_bias` a categorical value (i.e. strings, not numbers)? I also noticed that plottr crashes with this same stack trace when trying to plot categorical data.

**Comment by damazter** [comment]
`NW_bias` is not a categorical value, usually when this crash occurs, I can just restart plottr and plot the same Dataset without trouble

**Comment by mgunyho** [comment]
Oh, it is a different issue then

**Comment by wpfff** [comment]
After you showed me in the lab, i think this might be a case where we try to load data at an 'unfortunate' time (i.e., when data in the database has a weird shape while a writing step isn't quite finished yet). I'll try to reproduce and check if it's that.

**Comment by damazter** [comment]
Doing more tests in the lab, it turns out that we have to be very fortunate for the data to load while the measurement is running to a point where lifeplotting becomes completely impossible. So if you need help reproducing the issue, drop by the setup again :)

**Comment by wpfff** [comment]
lol. hopefully tomorrow!

**Comment by wpfff** [comment]
This was the same bug reported in #83 - this is fixed now (PR #96 ).
